notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2023-10-26 00:58:30.700274 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<Response [200]>"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "print_timestamp()\n",
+    "datagen_request = {\n",
+    "    \"name\": \"users\",\n",
+    "    \"target\": { \"type\": \"file\", \"path\":\"users.json\"  },\n",
+    "    \"config_file\": \"clickstream/users_init.json\", \n",
+    "    \"concurrency\":4000,\n",
+    "    \"total_events\":4000 \n",
+    "}\n",
+    "datagen.post(\"/start\", json.dumps(datagen_request), headers=headers)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "2beda33c-6f22-456d-9189-681d547cca8a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"name\": \"users\",\n",
+      "  \"config_file\": \"clickstream/users_init.json\",\n",
+      "  \"target\": {\n",
+      "    \"type\": \"file\",\n",
+      "    \"path\": \"/files/users.json\"\n",
+      "  },\n",
+      "  \"active_sessions\": 0,\n",
+      "  \"total_records\": 4000,\n",
+      "  \"start_time\": \"2023-10-26 00:58:30\",\n",
+      "  \"run_time\": 659.275399,\n",
+      "  \"status\": \"COMPLETE\",\n",
+      "  \"status_msg\": \"Running, Sim Clock: 2023-10-26 01:09:29.980941\"\n",
+      "}\n",
+      "2023-10-26 00:58:31.830364 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 0, 58, 31, 830364)"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "wait_for_datagen( \"users\")\n",
+    "print_timestamp()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ddd8faf9-a267-4607-83a0-85c7720c39cc",
+   "metadata": {},
+   "source": [
+    "### Load `clicks` and `users` Into Separate Tables\n",
+    "\n",
+    "Load separate tables for `clicks` and `users` to demonstrate join funtionality."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a819cf3-1403-42d4-81de-fbf2991c6c9e",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true
+   },
+   "source": [
+    "#### Generate Ingestion SQL from Data Files\n",
+    "While this notebook already includes the ingestion SQL, here are the details on how to generate it using the [Druid Console](http://localhost:8888/unified-console.html#):"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5443904f-6abb-4fb9-aec3-39d564de9f06",
+   "metadata": {},
+   "source": [
+    "Instead of manually figuring out the EXTERN table syntax needed to build the ingestion SQL statement, you can generate the SQL statements from the data file by using the Druid Console and following these steps:\n",
```

Comment:
I'd remove this section - dropping people out into the Druid Console is distracting.

Reply:
done